packages/@lwc/ssr-runtime/src/index.ts

```diff
@@ -172,20 +172,32 @@ export class LightningElement implements PropsAvailableAtConstruction {
         return (this.__classList = new ClassList(this));
     }
 
-    getAttribute(attrName: string): string | null {
-        return this.__attrs[attrName] ?? null;
+    #setAttribute(attrName: string, attrValue: string | null): void {
+        this.__attrs[attrName] = attrValue;
     }
 
-    setAttribute(attrName: string, value: string | null): void {
-        this.__attrs[attrName] = String(value);
+    setAttribute(attrName: string, attrValue: unknown): void {
+        this.#setAttribute(attrName, String(attrValue));
```

Comment:
It's a bit confusing to have two methods named `setAttribute` and `#setAttribute`. Maybe `#setAttributeWithoutTypeCoercion` or something?